### PR TITLE
Metadata updated for ClusterCreate API

### LINKED ISSUE #450: Populate ClusterCreate docs
Documentation here https://conbench.ursa.dev/api/docs/#/ under `ClusterCreate` has no metadata populated yet:

image

These fields describe clusters along these lines (though you might need to edit these slightly to fit with the style of the other API docs and to shorten them a bit):
- `info`: Information on the cluster (e.g. hosts, nodes or number of workers) that has been configured to run a set of benchmarks. This information should include all cluster configuration details that we suspect might impact benchmark performance (types of hardware for the nodes, number of nodes, etc). This mirrors the tags in the Machine Info. This information will be used to separate the results from the same benchmark run on different sets of hardware.
- `name`: A descriptive name of a cluster. This should be short, but distinctive — it is displayed in the conbench web app UI
- `optional_info`: Any additional optional information about the cluster. This is a good place to put configuration details that we don't expect to impact benchmark performance (e.g. region, settings like logging type, etc.). This information should be about the cluster itself that is the same across benchmarks run on the same cluster (e.g. if you have information that is unique to each benchmark, that should go in `optional_benchmark_info`) 

This metadata can be added here: 
https://github.com/conbench/conbench/blob/093e61385f9716b801ec03a773cfe281c43ce4c8/conbench/entities/hardware.py#L190-L193

Much like they are populated here:
https://github.com/conbench/conbench/blob/fdb30f2934481989e981c49837506c0622d6bb3a/conbench/entities/run.py#L217-L222

### PR DESCRIPTION
**Pull request by sayantikabanik**
resolves #450

### CODE REVIEW

File: conbench/entities/hardware.py [hunk 1]
```diff
@@ -188,9 +188,27 @@ class MachineCreate(marshmallow.Schema):
 
 
 class ClusterCreate(marshmallow.Schema):
-    name = marshmallow.fields.String(required=True)
-    info = marshmallow.fields.Dict(required=True)
-    optional_info = marshmallow.fields.Dict(required=True)
+    name = marshmallow.fields.String(
+        required=True,
+        metadata={
+            "description": "Information related to cluster (e.g. `hosts`, `nodes` or `number of workers`)"
```
Comment (jgehrcke) [inline conbench/entities/hardware.py]: I think there is a trailing space required before the closing `"` in this line.

Not just here, in all of those lines where the sentence continues in the next line.

File: conbench/entities/hardware.py [hunk 2]
```diff
@@ -188,9 +188,27 @@ class MachineCreate(marshmallow.Schema):
...
+            "configured to run a set of benchmarks. Used to differentiate between similar"
+            "benchmark runs performed on different sets of hardware"
+        },
+    )
+    info = marshmallow.fields.Dict(
+        required=True,
+        metadata={
+            "description": "`Distinct` name of the cluster, to be displayed on the web UI."
```
Comment (jgehrcke) [inline conbench/entities/hardware.py]: Just curious: why is the word "Distinct" emphasized/marked as inline code?

**Review by jgehrcke — DISMISSED** [review]
Thank you!

I left two comments :)

(also thinking out loud: do we have or need/want super simple tests somewhere that cover in a very coarse-grained fashion that this help text is after all displayed? -- I would understand if we decide to simply rely on the entity definition / swagger pipeline to do the right thing)

File: conbench/entities/hardware.py [hunk 3]
```diff
@@ -188,9 +188,27 @@ class MachineCreate(marshmallow.Schema):
 
 
 class ClusterCreate(marshmallow.Schema):
-    name = marshmallow.fields.String(required=True)
-    info = marshmallow.fields.Dict(required=True)
-    optional_info = marshmallow.fields.Dict(required=True)
+    name = marshmallow.fields.String(
```
Comment (jonkeane) [inline conbench/entities/hardware.py:191]: I believe you've got the description for info here (and then the description for name down below, yeah?

```suggestion
    info = marshmallow.fields.String(
```

File: conbench/entities/hardware.py [hunk 4]
```diff
@@ -188,9 +188,27 @@ class MachineCreate(marshmallow.Schema):
...
+        required=True,
+        metadata={
+            "description": "Information related to cluster (e.g. `hosts`, `nodes` or `number of workers`)"
+            "configured to run a set of benchmarks. Used to differentiate between similar"
+            "benchmark runs performed on different sets of hardware"
+        },
+    )
+    info = marshmallow.fields.Dict(
```
Comment (jonkeane) [inline conbench/entities/hardware.py:197]: Only if you suggest my comment above — alternatively you can swap the descriptions if you want to keep the order of info, name, optional_info the same

```suggestion
    name = marshmallow.fields.Dict(
```

File: conbench/entities/hardware.py [hunk 5]
```diff
@@ -188,9 +188,27 @@ class MachineCreate(marshmallow.Schema):
...
+        required=True,
+        metadata={
+            "description": "`Distinct` name of the cluster, to be displayed on the web UI."
+        },
+    )
+    optional_info = marshmallow.fields.Dict(
+        required=True,
+        metadata={
+            "description": "Additional optional information about the cluster, which may not impact the"
```
Comment (jonkeane) [inline conbench/entities/hardware.py]: A very minor style edit (though feel free to disregard if you think it makes this overly complicated) 

```suggestion
            "description": "Additional optional information about the cluster, which is likely to not impact the"
```

**Review by jonkeane — CHANGES_REQUESTED** [review]
Thanks for this. This is a fantastic job so far. 

I agree with @jgehrcke that having a test might be nice. Though I'm a little surprised this wasn't caught with the test at https://github.com/conbench/conbench/blob/main/conbench/tests/api/test_docs.py [the fixtures used there](https://github.com/conbench/conbench/blob/c086edf17a451002d278fe10b837c5a06edd0db8/conbench/tests/api/_expected_docs.py#L1068-L1073) don't have any descriptions

File: conbench/entities/hardware.py (hunk 3 above)
Comment (sayantikabanik) [reply conbench/entities/hardware.py:191]: Thanks for the pointer, I have swapped the entries

File: conbench/entities/hardware.py (hunk 2 above)
Comment (sayantikabanik) [reply conbench/entities/hardware.py]: Based on the description of the issue under #450  

`name: A descriptive name of a cluster. This should be short, but distinctive — it is displayed in the conbench web app UI`

The keyword `distinctive` felt important hence the highlight there. 
(But) Maybe having a check would be better instead (if not already present)

File: conbench/entities/hardware.py (hunk 5 above)
Comment (sayantikabanik) [reply conbench/entities/hardware.py]: `likely` suggests a tone similar to `probably`, which sounds great.

**Comment by sayantikabanik** [comment]
Thanks @jgehrcke @jonkeane for the feedback, I have updated the suggested changes :) 

Few observations:
- The test suite I believe expects a certain order of fields (eg:a,b,c) if swapped (eg:b,a,c)the test cases fails. Was wondering if a keyword match based approach could be helpful here.
- The metadata gets hidden in a fold-up where the field datatype is an `dictionary`. Is this a design choice?

**Comment by jonkeane** [comment]
Thanks for those fixes! I think this is really really close (the content is all good IMO). But one thing I'm surprised by is that I would have expected you to need to add those same descriptions to this file on these lines too:

https://github.com/conbench/conbench/blob/1ba4f354abdaf9b43fba34bdb4af528ad8ff51a2/conbench/tests/api/_expected_docs.py#L1069-L1072

That should be used by the docs test to confirm that we're seeing them. Would you mind trying to add them (and run the tests locally or use the CI to run the tests) and see if everything still passes? We don't _absolutely have to_ solve this mystery, but I expected some tests to fail without updating the `_expected_docs` fixture, so it would be good to figure out why that happened, IMO

**Comment by sayantikabanik** [comment]
In my local I see two tests failing (`test_login_page_shows_sso_link` & `test_oidc_flow_against_dex`), long/short summary below:

Both seems to be related to login (via web app)

<details>
<summary> Short version </summary>

```shell
============================= short test summary info======================================
FAILED conbench/tests/app/test_auth.py::TestLoginOIDC::test_login_page_shows_sso_link - assert 'Google Login' in '<!DOCTYPE html>\n<html>\n  <head>\n    <title>\n    Sign In - Conbench\n</title>\n    <meta name="viewport" content="width=device-width, initial-scale=1.0">\n\n    <!-- Bootstrap -->\n    <link href="//cdnjs.cloudflare.com/ajax/libs/twitte...
FAILED conbench/tests/app/test_auth.py::TestLoginOIDC::test_oidc_flow_against_dex - werkzeug.exceptions.BadRequestKeyError: 400 Bad Request: The browser (or proxy) sent a request that this server could not understand.
```
</details>

<details>
<summary> Logs (longer version) </summary>

```shell
================================================================================================================================== FAILURES ==================================================================================================================================
________________________________________________________________________________________________________________ TestLoginOIDC.test_login_page_shows_sso_link ________________________________________________________________________________________________________________

self = <conbench.tests.app.test_auth.TestLoginOIDC object at 0x1061b0730>, client = <FlaskClient <Flask 'conbench'>>

    def test_login_page_shows_sso_link(self, client):
        r = client.get("/login/", follow_redirects=True)
>       assert "Google Login" in r.text
E       assert 'Google Login' in '<!DOCTYPE html>\n<html>\n  <head>\n    <title>\n    Sign In - Conbench\n</title>\n    <meta name="viewport" content="width=device-width, initial-scale=1.0">\n\n    <!-- Bootstrap -->\n    <link href="//cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet">\n<link rel="shortcut icon" href="/static/favicon.ico?q=1669625722">\n<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" type="text/css" />\n<link href="https://cdn.datatables.net/1.10.23/css/dataTables.bootstrap.min.css" rel="stylesheet" type="text/css">\n<link href="https://cdn.datatables.net/responsive/2.2.7/css/responsive.bootstrap.min.css" rel="stylesheet" type="text/css">\n<link href="/static/app.css?q=1669625722" rel=stylesheet type=text/css />\n\n  </head>\n  <body>\n    \n    <nav class="navbar navbar-default">\n        <div class="container">\n            <div class="navbar-header">\n                <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">\n                    <span class="sr-only">Toggle navigation</span>\n                    <span class="icon-bar"></span>\n                    <span class="icon-bar"></span>\n                    <span class="icon-bar"></span>\n                </button>\n                <a class="navbar-brand" href="/">Conbench</a>\n            </div>\n            <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">\n                \n                    <ul class="nav navbar-nav navbar-right">\n                        <li><a href="/login/">Login</a></li>\n                    </ul>\n                \n            </div>\n        </div>\n    </nav>\n\n    <div class="modal fade" id="confirm-delete" tabindex="-1" role="dialog" aria-hidden="true">\n        <div class="modal-dialog">\n            <div class="modal-content">\n                <div class="modal-header">\n                    <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>\n                    <h4 class="modal-title"><span class="glyphicon glyphicon-trash submenu"></span>&nbsp; &nbsp; Confirm Delete</h4>\n                </div>\n                <div class="modal-body">\n                    <p>Are you sure?</p>\n                    <p id="confirm-delete-message"></p>\n                    <input id="delete-form-id" type="hidden" value="">\n                    <input id="delete-form-action" type="hidden" value="">\n                </div>\n                <div class="modal-footer">\n                    <button type="button" class="btn btn-default" data-dismiss="modal">Cancel</button>\n                    <button id="confirm-delete-button" type="button" class="btn btn-primary" data-dismiss="modal">Delete</button>\n                </div>\n            </div>\n        </div>\n    </div>\n\n    \n    <div class="container">\n        <br>\n        \n        \n            <div align="center">\n                <span class="flash-message"></span>\n                <br>\n            </div>\n        \n        \n\n        \n        \n<h1>Sign In</h1>\n<br/>\n<div class="row">\n\n<div class="col-md-4">\n  \n<form action="" method="post"\n  class="form" role="form">\n  <input id="csrf_token" name="csrf_token" type="hidden" value="IjQ5MjJkMThiMzA5ODE3ZTQ3ZTVmMmE3NzQ0ZmVhYTlhODc2MGY3Yzgi.Y4bccg.rQx9eJsLFp0-HwFs3pGejjCTp1c">\n  \n    \n\n\n\n\n\n\n<div class="form-group  required"><label class="control-label" for="email">Email</label>\n        \n          <input class="form-control" id="email" name="email" required type="text" value="">\n        \n  </div>\n\n\n    \n\n\n\n\n\n\n<div class="form-group  required"><label class="control-label" for="password">Password</label>\n        \n          <input class="form-control" id="password" name="password" required type="password" value="">\n        \n  </div>\n\n\n    \n\n\n\n\n\n  \n\n\n    <div class="checkbox">\n      <label>\n        <input id="remember_me" name="remember_me" type="checkbox" value="y"> Remember Me\n      </label>\n    </div>\n  \n\n\n\n\n    \n\n\n\n\n\n  \n\n  \n  \n\n\n    <input class="btn btn-default" id="submit" name="submit" type="submit" value="Submit">\n  \n\n\n\n\n    \n\n</form>\n  <br>\n  <p>New user? <a href="/register/">Sign Up</a></p>\n  \n</div>\n\n<div class="col-md-8">\n  <div class="jumbotron">\n    <div class="media">\n\n  <div class="media-body">\n    <h4 class="media-heading">Con·bench</h4>\n     <h4>/kənˈben(t)SH/</h4>\n     <h5><i>noun</i></h5>\n     <h3>Language-independent Continuous Benchmarking (CB) Framework</h3>\n  </div>\n  <div class="media-right">\n     <img class="media-object" style="padding: 5px" src="/static/bar-graph.png?q=1669625722" height="150">\n  </div>\n\n    </div>\n  </div>\n</div>\n\n</div>\n\n    </div>\n\n\n    \n<script type="text/javascript" src="https://code.jquery.com/jquery-3.5.1.js"></script>\n<script type="text/javascript" src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>\n<script type="text/javascript" src="https://cdn.datatables.net/1.10.23/js/jquery.dataTables.min.js"></script>\n<script type="text/javascript" src="https://cdn.datatables.net/1.10.23/js/dataTables.bootstrap.min.js"></script>\n<script type="text/javascript" src="https://cdn.datatables.net/responsive/2.2.7/js/dataTables.responsive.min.js"></script>\n<script type="text/javascript" src="https://cdn.datatables.net/responsive/2.2.7/js/responsive.bootstrap.min.js"></script>\n\n<script type="text/javascript">\n$(document).ready(function() {\n    setTimeout(function() {\n        $(".flash-message").fadeOut("slow");\n    }, 2000); // <-- time in milliseconds\n});\n\nfunction enable_search_query_string(data_table) {\n    $(data_table).on( \'search.dt\', function () {\n        var current_search_value = $(\'.dataTables_filter input\').val();\n        var new_url = window.location.protocol + "//" + window.location.host + window.location.pathname\n        if (current_search_value) {\n            new_url += "?search=" + encodeURIComponent(current_search_value)\n        }\n        window.history.replaceState(null, document.title, new_url)\n    });\n\n    $(document).ready(function() {\n        var table =  $(data_table).DataTable();\n        \n    });\n}\n\n$("#confirm-delete").on("show.bs.modal", function(e) {\n    $(this).find("#confirm-delete-message").html($(e.relatedTarget).data("message"));\n    $(this).find("#delete-form-id").attr("value", $(e.relatedTarget).data("form-id"));\n    $(this).find("#delete-form-action").attr("value", $(e.relatedTarget).data("href"));\n});\n\n$("#confirm-delete-button").click(function(e){\n    $($("#delete-form-id").attr("value")).find("#delete").attr("type", "submit");\n    $($("#delete-form-id").attr("value")).attr("action", $("#delete-form-action").attr("value"));\n    $("#delete").click();\n});\n</script>\n\n  </body>\n</html>\n'
E        +  where '<!DOCTYPE html>\n<html>\n  <head>\n    <title>\n    Sign In - Conbench\n</title>\n    <meta name="viewport" content="width=device-width, initial-scale=1.0">\n\n    <!-- Bootstrap -->\n    <link href="//cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet">\n<link rel="shortcut icon" href="/static/favicon.ico?q=1669625722">\n<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" type="text/css" />\n<link href="https://cdn.datatables.net/1.10.23/css/dataTables.bootstrap.min.css" rel="stylesheet" type="text/css">\n<link href="https://cdn.datatables.net/responsive/2.2.7/css/responsive.bootstrap.min.css" rel="stylesheet" type="text/css">\n<link href="/static/app.css?q=1669625722" rel=stylesheet type=text/css />\n\n  </head>\n  <body>\n    \n    <nav class="navbar navbar-default">\n        <div class="container">\n            <div class="navbar-header">\n                <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">\n                    <span class="sr-only">Toggle navigation</span>\n                    <span class="icon-bar"></span>\n                    <span class="icon-bar"></span>\n                    <span class="icon-bar"></span>\n                </button>\n                <a class="navbar-brand" href="/">Conbench</a>\n            </div>\n            <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">\n                \n                    <ul class="nav navbar-nav navbar-right">\n                        <li><a href="/login/">Login</a></li>\n                    </ul>\n                \n            </div>\n        </div>\n    </nav>\n\n    <div class="modal fade" id="confirm-delete" tabindex="-1" role="dialog" aria-hidden="true">\n        <div class="modal-dialog">\n            <div class="modal-content">\n                <div class="modal-header">\n                    <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>\n                    <h4 class="modal-title"><span class="glyphicon glyphicon-trash submenu"></span>&nbsp; &nbsp; Confirm Delete</h4>\n                </div>\n                <div class="modal-body">\n                    <p>Are you sure?</p>\n                    <p id="confirm-delete-message"></p>\n                    <input id="delete-form-id" type="hidden" value="">\n                    <input id="delete-form-action" type="hidden" value="">\n                </div>\n                <div class="modal-footer">\n                    <button type="button" class="btn btn-default" data-dismiss="modal">Cancel</button>\n                    <button id="confirm-delete-button" type="button" class="btn btn-primary" data-dismiss="modal">Delete</button>\n                </div>\n            </div>\n        </div>\n    </div>\n\n    \n    <div class="container">\n        <br>\n        \n        \n            <div align="center">\n                <span class="flash-message"></span>\n                <br>\n            </div>\n        \n        \n\n        \n        \n<h1>Sign In</h1>\n<br/>\n<div class="row">\n\n<div class="col-md-4">\n  \n<form action="" method="post"\n  class="form" role="form">\n  <input id="csrf_token" name="csrf_token" type="hidden" value="IjQ5MjJkMThiMzA5ODE3ZTQ3ZTVmMmE3NzQ0ZmVhYTlhODc2MGY3Yzgi.Y4bccg.rQx9eJsLFp0-HwFs3pGejjCTp1c">\n  \n    \n\n\n\n\n\n\n<div class="form-group  required"><label class="control-label" for="email">Email</label>\n        \n          <input class="form-control" id="email" name="email" required type="text" value="">\n        \n  </div>\n\n\n    \n\n\n\n\n\n\n<div class="form-group  required"><label class="control-label" for="password">Password</label>\n        \n          <input class="form-control" id="password" name="password" required type="password" value="">\n        \n  </div>\n\n\n    \n\n\n\n\n\n  \n\n\n    <div class="checkbox">\n      <label>\n        <input id="remember_me" name="remember_me" type="checkbox" value="y"> Remember Me\n      </label>\n    </div>\n  \n\n\n\n\n    \n\n\n\n\n\n  \n\n  \n  \n\n\n    <input class="btn btn-default" id="submit" name="submit" type="submit" value="Submit">\n  \n\n\n\n\n    \n\n</form>\n  <br>\n  <p>New user? <a href="/register/">Sign Up</a></p>\n  \n</div>\n\n<div class="col-md-8">\n  <div class="jumbotron">\n    <div class="media">\n\n  <div class="media-body">\n    <h4 class="media-heading">Con·bench</h4>\n     <h4>/kənˈben(t)SH/</h4>\n     <h5><i>noun</i></h5>\n     <h3>Language-independent Continuous Benchmarking (CB) Framework</h3>\n  </div>\n  <div class="media-right">\n     <img class="media-object" style="padding: 5px" src="/static/bar-graph.png?q=1669625722" height="150">\n  </div>\n\n    </div>\n  </div>\n</div>\n\n</div>\n\n    </div>\n\n\n    \n<script type="text/javascript" src="https://code.jquery.com/jquery-3.5.1.js"></script>\n<script type="text/javascript" src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>\n<script type="text/javascript" src="https://cdn.datatables.net/1.10.23/js/jquery.dataTables.min.js"></script>\n<script type="text/javascript" src="https://cdn.datatables.net/1.10.23/js/dataTables.bootstrap.min.js"></script>\n<script type="text/javascript" src="https://cdn.datatables.net/responsive/2.2.7/js/dataTables.responsive.min.js"></script>\n<script type="text/javascript" src="https://cdn.datatables.net/responsive/2.2.7/js/responsive.bootstrap.min.js"></script>\n\n<script type="text/javascript">\n$(document).ready(function() {\n    setTimeout(function() {\n        $(".flash-message").fadeOut("slow");\n    }, 2000); // <-- time in milliseconds\n});\n\nfunction enable_search_query_string(data_table) {\n    $(data_table).on( \'search.dt\', function () {\n        var current_search_value = $(\'.dataTables_filter input\').val();\n        var new_url = window.location.protocol + "//" + window.location.host + window.location.pathname\n        if (current_search_value) {\n            new_url += "?search=" + encodeURIComponent(current_search_value)\n        }\n        window.history.replaceState(null, document.title, new_url)\n    });\n\n    $(document).ready(function() {\n        var table =  $(data_table).DataTable();\n        \n    });\n}\n\n$("#confirm-delete").on("show.bs.modal", function(e) {\n    $(this).find("#confirm-delete-message").html($(e.relatedTarget).data("message"));\n    $(this).find("#delete-form-id").attr("value", $(e.relatedTarget).data("form-id"));\n    $(this).find("#delete-form-action").attr("value", $(e.relatedTarget).data("href"));\n});\n\n$("#confirm-delete-button").click(function(e){\n    $($("#delete-form-id").attr("value")).find("#delete").attr("type", "submit");\n    $($("#delete-form-id").attr("value")).attr("action", $("#delete-form-action").attr("value"));\n    $("#delete").click();\n});\n</script>\n\n  </body>\n</html>\n' = <WrapperTestResponse 6740 bytes [200 OK]>.text

conbench/tests/app/test_auth.py:145: AssertionError
----------------------------------------------------------------------------------------------------------------------------- Captured log setup -----------------------------------------------------------------------------------------------------------------------------
DEBUG    conbench.db:before.py:35 Starting call to 'conbench.db.create_all', this is the 1st time calling it.
DEBUG    conbench:__init__.py:66 Config.CREATE_ALL_TABLES appears to be set, call create_all()
DEBUG    conbench.db:before.py:35 Starting call to 'conbench.db.create_all', this is the 1st time calling it.
DEBUG    conbench:__init__.py:30 re-configure logging
__________________________________________________________________________________________________________________ TestLoginOIDC.test_oidc_flow_against_dex __________________________________________________________________________________________________________________

self = <conbench.tests.app.test_auth.TestLoginOIDC object at 0x1061b08e0>, client = <FlaskClient <Flask 'conbench'>>

    def test_oidc_flow_against_dex(self, client):
    
        # TODO: parse this 'initiate flow URL' from the HTML login page, i.e.
        # from the button/ link that people would actually click. If that is a
        # _relative_ href then construct an absolute URL from it (see below).
    
        # As long as `client` is the Flask test client we do not use an actual
        # HTTP client/server interaction. The URL here must be specified in
        # absolute terms because the request handler uses scheme, host, port to
        # dynamically generate the OIDC callback URL. We could set any scheme
        # and host, but we want to match the allow-listed callback URL that Dex
        # is aware of. That is is currently set to
        # http://127.0.0.1:5000/api/google/callback -- see
        # containders/dex/config.yml
        r0 = client.get("http://127.0.0.1:5000/api/google/")
    
        # `r0` is meant to be a redirect response, redirecting to the identity
        # provider. Extract the full URL we've been redirected to. The URL
        # represents a so-called authorization request, with all the parameters
        # for that request being encoded in the URL query parameters
>       authorization_request_url = r0.headers["location"]

conbench/tests/app/test_auth.py:167: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Headers([('Content-Type', 'application/json'), ('Content-Length', '46')]), key = 'location', _get_mode = False

    def __getitem__(self, key, _get_mode=False):
        if not _get_mode:
            if isinstance(key, int):
                return self._list[key]
            elif isinstance(key, slice):
                return self.__class__(self._list[key])
        if not isinstance(key, str):
            raise exceptions.BadRequestKeyError(key)
        ikey = key.lower()
        for k, v in self._list:
            if k.lower() == ikey:
                return v
        # micro optimization: if we are in get mode we will catch that
        # exception one stack level down so we can raise a standard
        # key error instead of our special one.
        if _get_mode:
            raise KeyError()
>       raise exceptions.BadRequestKeyError(key)
E       werkzeug.exceptions.BadRequestKeyError: 400 Bad Request: The browser (or proxy) sent a request that this server could not understand.

../../envs/conbench/lib/python3.9/site-packages/werkzeug/datastructures.py:907: BadRequestKeyError
----------------------------------------------------------------------------------------------------------------------------- Captured log setup -----------------------------------------------------------------------------------------------------------------------------
DEBUG    conbench.db:before.py:35 Starting call to 'conbench.db.create_all', this is the 1st time calling it.
DEBUG    conbench:__init__.py:66 Config.CREATE_ALL_TABLES appears to be set, call create_all()
DEBUG    conbench.db:before.py:35 Starting call to 'conbench.db.create_all', this is the 1st time calling it.
DEBUG    conbench:__init__.py:30 re-configure logging
---------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------
[221130-10:00:42.907] [31275] [conbench] ERROR: Exception on /api/google/ [GET]
Traceback (most recent call last):
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/views.py", line 107, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/views.py", line 188, in dispatch_request
    return current_app.ensure_sync(meth)(**kwargs)
  File "/Users/sayantikabanik/workplace_vd/conbench/conbench/api/auth.py", line 77, in get
    return f.redirect(_google.auth_google_user())
  File "/Users/sayantikabanik/workplace_vd/conbench/conbench/api/_google.py", line 79, in auth_google_user
    client, oidc_provider_config = get_oidc_client()
  File "/Users/sayantikabanik/workplace_vd/conbench/conbench/api/_google.py", line 23, in get_oidc_client
    discovery_url, client_id, _ = get_oidc_config()
  File "/Users/sayantikabanik/workplace_vd/conbench/conbench/api/_google.py", line 16, in get_oidc_config
    discovery_url = Config.OIDC_ISSUER_URL + "/.well-known/openid-configuration"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
----------------------------------------------------------------------------------------------------------------------------- Captured log call ------------------------------------------------------------------------------------------------------------------------------
ERROR    conbench:app.py:1741 Exception on /api/google/ [GET]
Traceback (most recent call last):
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/views.py", line 107, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
  File "/Users/sayantikabanik/envs/conbench/lib/python3.9/site-packages/flask/views.py", line 188, in dispatch_request
    return current_app.ensure_sync(meth)(**kwargs)
  File "/Users/sayantikabanik/workplace_vd/conbench/conbench/api/auth.py", line 77, in get
    return f.redirect(_google.auth_google_user())
  File "/Users/sayantikabanik/workplace_vd/conbench/conbench/api/_google.py", line 79, in auth_google_user
    client, oidc_provider_config = get_oidc_client()
  File "/Users/sayantikabanik/workplace_vd/conbench/conbench/api/_google.py", line 23, in get_oidc_client
    discovery_url, client_id, _ = get_oidc_config()
  File "/Users/sayantikabanik/workplace_vd/conbench/conbench/api/_google.py", line 16, in get_oidc_config
    discovery_url = Config.OIDC_ISSUER_URL + "/.well-known/openid-configuration"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str
```
</details>

**Comment by jgehrcke** [comment]
> In my local I see two tests failing (test_login_page_shows_sso_link & test_oidc_flow_against_dex), long/short summary below:

Hey @sayantikabanik did you invoke tests locally via docker-compose? I've just opened this README PR: https://github.com/conbench/conbench/pull/468 -- can you try the method that's documented there and see if that results in all tests passing on your machine?

**Comment by austin3dickey** [comment]
>I would have expected you to need to add those same descriptions to this file

I solved the mystery; see #471

**Comment by jonkeane** [comment]
I'm going to merge this, now that the mystery is solved. Though of course, feel free to continue the discussion about the docker invocation after the merge if that's helpful for learning purposes!

**Review by jonkeane — APPROVED** [review]

**Pull request merged**